test/errorshow.jl

```diff
@@ -1090,3 +1090,15 @@ let e = @test_throws MethodError convert(TypeCompareError{Float64,1}, TypeCompar
     @test  occursin("TypeCompareError{Float64,1}", str)
     @test !occursin("TypeCompareError{Float64{},2}", str) # No {...} for types without params
 end
+
+@testset "InexactError for Inf16 should print '16' (#51087)" begin
+    for IntType in [Int8, Int16, Int32, Int64, Int128, UInt8, UInt16, UInt32, UInt64, UInt128]
+        IntStr = string(IntType)
+        for InfVal in [Inf, Inf16, Inf32, Inf64]
```

Comment:
Array constructors promote their elements, so this is equivalent to `fill(Inf, 4)`
```suggestion
        for InfVal in Any[Inf, Inf16, Inf32, Inf64]
```